File: java-to-kotlin/src/main/kotlin/lec07/Lec07Main.java
package lec07;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

public class Lec07Main {

    public static void main(String[] args) throws IOException {

    }

    private Integer parseIntOrThrow(@NotNull String str) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            throw new IllegalArgumentException(String.format("주어진 %s는 숫자가 아닙니다.", str));
        }
    }

    private Integer parseIntOrThrow2(@NotNull String str) {
        try {
            return Integer.parseInt(str);
        } catch (NumberFormatException e) {
            return null;
        }
    }
}

File: java-to-kotlin/src/main/kotlin/lec07/Lec07Main.kt
package lec07

import java.lang.NumberFormatException

// 주어진 문자열을 정수로 변경하는 예제
fun parseIntOrThrow(str: String): Int {
    try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        throw IllegalArgumentException("주어진 ${str}는 숫자가 아닙니다")
    }
}

fun parseIntOrThrow2(str: String): Int? {
    return try {
        return str.toInt()
    } catch (e: NumberFormatException) {
        null
    }
}
